### PR TITLE
[BACKLOG-3385] - Upgrade karaf and its dependencies to 3.0.3 release

### PR DESCRIPTION
**Pull request by rmansoor**

### CODE REVIEW

**Comment by rmansoor** [comment]
@CodeOnCoffee  please review

File: pentaho-karaf-features/src/main/resources/pentaho-karaf-features-standard.xml [hunk 1]
```diff
@@ -52,20 +52,6 @@
 
 
   <!-- Overriding standard one to exclude log bundle which will not work without pax-logging -->
-  <feature name="management" description="Provide a JMX MBeanServer and a set of MBeans in Karaf" version="5.0">
```
Comment (pentaho-nbaker) [inline pentaho-karaf-features/src/main/resources/pentaho-karaf-features-standard.xml]: Why was this removed? We need to override this in order to exclude logging.

File: pentaho-karaf-features/src/main/resources/pentaho-karaf-features-standard.xml [hunk 2]
```diff
@@ -82,8 +68,362 @@
     <bundle start-level="90">blueprint:mvn:pentaho/pentaho-blueprint-activators/${project.version}/xml/camel-jms
     </bundle>
   </feature>
+  
+    <!-- Spring 2 support -->
+    <feature name="spring" description="Spring 2.x support" version="2.5.6.SEC03" resolver="(obr)">
```
Comment (pentaho-nbaker) [inline pentaho-karaf-features/src/main/resources/pentaho-karaf-features-standard.xml]: Were you unable to find a public feature file we can reference instead? I'd rather that than copy in all of these.

File: pentaho-karaf-features/src/main/resources/pentaho-karaf-features-standard.xml (hunk 1 above)
Comment (rmansoor) [inline pentaho-karaf-features/src/main/resources/pentaho-karaf-features-standard.xml]: I was getting errors when I included this feature. May be I can try this again and see what happens

File: pentaho-karaf-features/src/main/resources/pentaho-karaf-features-standard.xml (hunk 2 above)
Comment (rmansoor) [inline pentaho-karaf-features/src/main/resources/pentaho-karaf-features-standard.xml]: I searched but did not fine a public spring feature for 3.0.3. That is when I copied everything to our standards file

**Comment by rmansoor** [comment]
@CodeOnCoffee updated the pull request
